Code/Framework/AzCore/AzCore/Memory/PoolSchema.cpp

```diff
@@ -163,7 +163,7 @@ namespace AZ
         }
 
         using AllocatorType = PoolAllocation<PoolSchemaImpl>;
-        IAllocatorAllocate*                 m_pageAllocator;
+        IAllocatorSchema*                   m_pageAllocator;
```

Comment:
This could be queried from the `m_allocator` member on the `PoolAllocation` class to reduce a memberthe `m_pageAllocator`

Reply:
`m_allocator` is actually a wrapper of this class, so calling `m_allocator->Allocate` ends up calling this method. This `PoolSchemaImpl` is somehow a "schema" that is implemented through `PoolAllocation<>`
These pool allocators are not following the same class hierarchy structure that the rest of the allocators... needs a rewrite, but since I also want to simplify the allocator's class hierarchy, will hopefully tackle all that together in a future pass